### ISSUE TITLE
Type recursively references itself as a base type

### ISSUE THREAD

**Issue by lars-reimann** [eclipse-langium/langium#1182]
Langium 2.0.1 checks for cycles in the inheritance chains of interfaces that are defined in the grammar. However, it seems to flag all cases where the same interface can be reached via multiple paths, even if there is no cycle.

![image](https://github.com/eclipse-langium/langium/assets/2501322/70f84b48-f2cc-4a7e-a0b2-df7c0f8eae05)

Langium version: 2.0.1
Langium-CLI version: 2.0.1
Langium VSC-Extension version: 2.0.0
Package name: Langium/Langium-CLI

## Steps To Reproduce

1. Use the following grammar in any Langium project (e.g. "hello-world"):
```
grammar HelloWorld

interface Declaration {
    name: string
}

interface ModuleMember extends Declaration {}
interface ClassMember extends Declaration {}
interface Func extends ModuleMember, ClassMember {}

entry Model:
    (persons+=Person)*;

Person:
    'person' name=ID;

hidden terminal WS: /\s+/;
terminal ID: /[_a-zA-Z][\w_]*/;
```

## The current behavior

An error is raised with a misleading message and `langium:generate` no longer works.

## The expected behavior

In my opinion, no error should be raised in this case and generation should work. At the very least, the error message should be adjusted.

**Comment by msujew** [comment]
This has been resolved in https://github.com/eclipse-langium/langium/pull/1175, release will follow shortly.

**Comment by lars-reimann** [comment]
Ah sorry, I was just searching for issues not PRs. Thanks for the fix.

**Comment by msujew** [comment]
The fix has been released in [2.0.2](https://www.npmjs.com/package/langium/v/2.0.2) 🎉
